app.rb

```diff
@@ -0,0 +1,155 @@
+      puts 'Sorry, there is no one, add your friends'
+    else
+      @people.each_with_index do |person, index|
+        puts "#{show_index ? index : ' '} [#{person.class}]" \
+             "Name: \"#{person.name.delete("\n")}\", ID: #{person.id}, Age: #{person.age}"
+      end
+    end
+    puts
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    person_type = gets.chomp.to_i
+    case person_type
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      'Invalid number, try again'
+    end
+  end
+
+  def create_student
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp
+    print 'Class: '
+    classroom = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    permission = gets.chomp.downcase == 'y' || gets.chomp.downcase == ' '
+    student = Student.new(age, classroom, name: name, parent_permission: permission)
+    @people << student
+    puts 'Person created successfully'
+    puts ' '
+  end
+
+  def create_teacher
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(age, specialization, name: name)
+    @people << teacher
+    puts 'Person Created Successfully'
+    puts ' '
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    book = Book.new(title, author)
+    @books << book
+    puts 'Book Created Successfully'
+    puts ' '
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by number'
+    book_list(show_index: true)
+    book_index = gets.chomp.to_i
+    puts 'Select a person from the following list by number (not id)'
+    person_list(show_index: true)
+    person_index = gets.chomp.to_i
+    print 'Date (dd/mm/yy): '
+    date = Date.parse(gets.chomp)
+    rental = Rental.new(date, @books[book_index], @people[person_index])
+    @rentals << rental
+    puts 'Rental Created Successfully'
+    puts ' '
+  end
+
+  def list_rentals
+    print 'ID of person: '
+    id = gets.chomp.to_i
+    puts 'Rentals:'
+    @rentals.each do |rental|
+      puts "Date: #{rental.date}, Book: #{rental.book.title} by #{rental.book.author}" if rental.person.id == id
+      puts ' '
+    end
+  end
+
+  def menu_list
+    puts 'Please select an option by entering a number:'
+    puts '1. List all books'
+    puts '2. List all people'
+    puts '3. Create a person'
+    puts '4. Create a book'
+    puts '5. Create a book rental'
+    puts '6. List all rentals for a given person id '
+    puts '7. Exit'
+  end
+
+  def menu_action(option)
+    case option
+    when 1
+      book_list
+    when 2
+      person_list
+    when 3
+      create_person
+    when 4
+      create_book
+    when 5
+      create_rental
+    when 6
+      list_rentals
+    else
+      'Invalid option, try again'
+    end
```

Comment:
## Recommendation:

- [ ] It would be nice if you could add option 7 and when chosen the user is then thanked for using the app. Feedback like - "Thank you for using this app!" 👍🏽